hashintel/hash
# Add new entity (from scratch) creation flow

## 🌟 What is the purpose of this PR?

This PR implements the entity creation flow, using this url: `/@{shortname}/entities/new`, and makes `create entity` button at entity-type-editor functional.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/0/1203203788735441) _(internal)_
- [Design](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-App?node-id=6067%3A248022&t=OX8sb5WsLKVk6joK-4) _(internal)_

## 🔍 What does this change?

- Improve `generateEntityLabel`
- Add `isPropertyValueArray` & `isPropertyNested` type guards
- Implement create new entity page
- Implement create new entity with pre-selected type
- Extract empty states of property editor sections to separate files (to reuse them)


## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Open `http://localhost:3000/@{shortname}/entities/new`
1. Select an entity type from dropdown, and confirm entity creation is working
2. Open the entity type editor for an entity type
3. Click on `create new entity` button on the top right
4. Confirm creating an entity with a pre-selected type works

## 📹 Demo

https://user-images.githubusercontent.com/39553853/203777019-7a9f6c7d-6353-4a3a-90f9-3492bfb2ca9e.mov
